sdb/commands/linux/slabs.py

```diff
@@ -0,0 +1,141 @@
+#
+# Copyright 2019 Delphix
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+import argparse
+from typing import Any, Callable, Dict, Iterable, List
+
+import drgn
+
+import sdb
+from sdb.commands.internal.fmt import size_nicenum
+from sdb.commands.internal.table import Table
+from sdb.commands.linux.internal import slub_helpers as slub
+
+
+class Slabs(sdb.Locator, sdb.PrettyPrinter):
+    # pylint: disable=too-few-public-methods
+
+    names = ["slabs"]
+
+    input_type = "struct kmem_cache *"
+    output_type = "struct kmem_cache *"
+
+    def _init_argparse(self, parser: argparse.ArgumentParser) -> None:
+        parser.add_argument(
+            '-H',
+            action='store_false',
+            help=
+            'do not display headers and separate fields by a single tab (scripted mode)'
+        )
+        parser.add_argument(
+            '-o',
+            metavar="PROPERTIES",
+            help='comma-separated list of properties to display')
+        parser.add_argument('-p',
+                            action='store_true',
+                            help='display numbers in parseable (exact) values')
+        parser.add_argument(
+            '-r',
+            '--recursive',
+            action='store_true',
+            help='recurse down children caches and print statistics')
+        parser.add_argument('-s',
+                            metavar="PROPERTY",
+                            help='sort rows by PROPERTY')
+        parser.add_argument('-v',
+                            action='store_true',
+                            help='Print all statistics')
+        parser.epilog = "PROPERTIES := {:s}".format(','.join(
+            Slabs.PROPERTIES.keys()))
```

Comment:
Not a big deal, but where the newline is placed here is a little awkward:
```
PROPERTIES := address,name,object_size,entry_size,entries_per_slab,slabs,mem_u
sed,objs,active_objs,inactive_objs,util
```
i.e. `mem_used` is split across multiple lines.